server/src/internalClusterTest/java/org/opensearch/snapshots/SegmentReplicationSnapshotIT.java

```diff
@@ -147,6 +150,54 @@ public void testRestoreOnSegRep() throws Exception {
         assertHitCount(resp, DOC_COUNT);
     }
 
+    public void testCreateSnapshotOnNewlyElectedPrimaryBeforePrimaryMode() throws Exception {
+        // Start cluster with one primary and one replica node
+        List<String> nodes = startClusterWithSettings(segRepEnableIndexSettings(), 1);
+        // drop the primary - will be first in this list.
+        // Snapshot declaration
+        Path absolutePath = randomRepoPath().toAbsolutePath();
+        createRepository(REPOSITORY_NAME, "fs", absolutePath);
+        final String primary = nodes.get(0);
+        final String replica = nodes.get(1);
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(primary));
+        // Create snapshot - this will occasionally fail because the replica has not completed promotion.
```

Comment:
To make it deterministic, is it possible to block the `updateShardState`  [transport call](https://github.com/opensearch-project/OpenSearch/blob/81c7b9773cbb862bbb3e36695077c62bcfdaa7b6/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L621) from cluster manager to elected replica shard. Please correct me if this action is not invoked on promoted replica.

Reply:
@dreamer-89 finally got around to this, added a unit test instead.

Reply:
Thanks @mch2 for the effort on adding unit test.